### PR TITLE
Add mistral WF example for rollback and retry

### PR DESCRIPTION
**Pull request by m4dcoder**

### CODE REVIEW

File: contrib/examples/actions/mistral-handle-retry.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+                action: core.local
+                input:
+                    cmd: "if [ ! -e \"/tmp/done\" ]; then exit 1; fi"
+                on-error:
+                    - undo
+            undo:
+                action: core.local
+                input:
+                    cmd: "echo \"File does not exist.\""
+                on-success:
```
Comment (dzimine) [inline contrib/examples/actions/mistral-handle-retry.yaml]: `on-complete`, I think.

File: contrib/examples/actions/mistral-handle-retry.yaml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+    main:
+        type: direct
+        tasks:
+            task1:
+                workflow: work
+                policies:
+                    retry:
+                        count: 20
+                        delay: 3
```
Comment (dzimine) [inline contrib/examples/actions/mistral-handle-retry.yaml]: would be good to use break-on, but I can't come up with idea how.

File: contrib/examples/actions/mistral-handle-retry.yaml [hunk 3]
```diff
@@ -0,0 +1,30 @@
+version: '2.0'
+name: "examples.mistral-handle-retry"
+
+workflows:
+
+    main:
+        type: direct
+        tasks:
+            task1:
```
Comment (dzimine) [inline contrib/examples/actions/mistral-handle-retry.yaml]: To make it fully automated, consider a parallel task with delay, which creates the file there.

File: contrib/examples/actions/mistral-handle-retry.yaml (hunk 1 above)
Comment (m4dcoder) [inline contrib/examples/actions/mistral-handle-retry.yaml]: yeah, the thing is what if the undo failed...  right now, the whole thing retries regardless.   so the undo WF must be pretty reliable.

File: contrib/examples/actions/mistral-handle-retry.yaml [hunk 4]
```diff
@@ -0,0 +1,30 @@
+version: '2.0'
+name: "examples.mistral-handle-retry"
+
+workflows:
+
```
Comment (dzimine) [inline contrib/examples/actions/mistral-handle-retry.yaml]: I would use description to tell the story of what the workflow is doing.

File: contrib/examples/actions/mistral-handle-retry.yaml (hunk 3 above)
Comment (dzimine) [inline contrib/examples/actions/mistral-handle-retry.yaml]: task1 -> test_retry?

File: contrib/examples/actions/mistral-handle-retry.yaml (hunk 2 above)
Comment (m4dcoder) [inline contrib/examples/actions/mistral-handle-retry.yaml]: i tried break-on.  the thing is the retry policy don't have an option to run an undo.

**Comment by dzimine** [comment]
Looks good. few comments, up to you.

File: contrib/examples/actions/mistral-handle-retry.yaml (hunk 3 above)
Comment (m4dcoder) [inline contrib/examples/actions/mistral-handle-retry.yaml]: good idea

File: contrib/examples/actions/mistral-handle-retry.yaml (hunk 1 above)
Comment (m4dcoder) [inline contrib/examples/actions/mistral-handle-retry.yaml]: maybe i'll add a break-on if the undo fails.

Comment (m4dcoder) [inline contrib/examples/actions/mistral-handle-retry.yaml]: a bit hackish though...

**Pull request merged**